cumulusci/cli/cci.py

```diff
@@ -216,25 +165,68 @@ def new_func(*args, **kw):
         return decorate(func)
 
 
-# Root command
+SUGGEST_GIT_GIST_COMMAND = """\nIt looks like you may have run into error. Did you know cci has a command for sending this error to a GitHub gist?
+Just run `$ cci gist` and make sure that your GitHub access token as the 'create gist' scope.\n"""
 
 
-@click.group("main", help="")
-def main():
+#
+# Root command
+#
+def main(args=None):
     """Main CumulusCI CLI entry point.
 
     This runs as the first step in processing any CLI command.
+
+    This wraps the `click` library in order to do some initialization and centralized error handling.
     """
-    # Avoid checking for updates if we've been asked to output JSON,
-    # or if we're going to check anyway as part of the `version` command.
-    is_version_command = len(sys.argv) > 1 and sys.argv[1] == "version"
-    if "--json" not in sys.argv and not is_version_command:
-        check_latest_version()
-    log_requests = "--debug" in sys.argv
-    init_logger(log_requests=log_requests)
+    with contextlib.ExitStack() as stack:
+        args = args or sys.argv
+        # Check for updates _unless_ we've been asked to output JSON,
+        # or if we're going to check anyway as part of the `version` command.
+        is_version_command = len(args) > 1 and args[1] == "version"
+        if "--json" not in args and not is_version_command:
+            check_latest_version()
+
+        # Load CCI config
+        global RUNTIME
+        RUNTIME = CliRuntime()
+        RUNTIME.check_cumulusci_version()
+
+        # Configure logging
+        debug = "--debug" in args
+        if debug:
+            args.remove("--debug")
+
+        # Only create logfiles for commands
+        # that are not `cci gist`
+        is_gist_command = len(args) > 1 and args[1] == "gist"
+        if not is_gist_command:
+            logger = get_gist_logger(RUNTIME.project_config.repo_root)
```

Comment:
This breaks when running a command outside of a project directory, such as `cci project init`:

```
Traceback (most recent call last):
  File "/Users/dglick/.pyenv/versions/cci/bin/cci", line 11, in <module>
    load_entry_point('cumulusci', 'console_scripts', 'cci')()
  File "/Users/dglick/Work/CumulusCI/cumulusci/cli/cci.py", line 204, in main
    logger = get_gist_logger(RUNTIME.project_config.repo_root)
AttributeError: 'NoneType' object has no attribute 'repo_root'
```

I'm actually reconsidering my earlier suggestion that logs should go inside the project directory. I don't see any particular problem with just having a central location for logs in `~/.cumulusci/logs`